utils/pre-publish-check.js

```diff
@@ -16,39 +18,55 @@ glob(
             const {publishConfig, name} = pkgJson;
 
             if (!publishConfig || publishConfig.access !== "public") {
-                // eslint-disable-next-line no-console
                 console.error(
                     `ERROR: ${name} is missing a "publishConfig": {"access": "public"} section.`,
                 );
                 process.exit(1);
             }
 
+            if (pkgJson.main !== "dist/index.js") {
+                console.error(
+                    `ERROR: ${name} must have a "main" set to "dist/index.js".`,
+                );
+                process.exit(1);
+            }
```

Comment:
I think all these error checks will be way easier to maintain if they were methods. The method names would document the purpose of the check, rather than relying on maintainers to insinuate the purpose from the implementation.

Reply:
something like:
```
checkPublishConfig();
checkMainField();
...
```
?